From 8d3e596314f3825dbde4052a12d193353dca8f8e Mon Sep 17 00:00:00 2001
From: zou cao <cao.zou@windriver.com>
Date: Mon, 4 Dec 2017 16:56:02 +0800
Subject: [PATCH] ARM: imx7d: clk: add init clk OCOTP

ocotp is used by other drivers in shadow registers address map,
normally it is enabled in bootloader, but in some cases without
bootloader, like kexec, ocotp will be disabled when system booting,
it will cause the cpu freeze when reading these shadow registers,
so enable it in system startup.

Signed-off-by: zou cao <cao.zou@windriver.com>
---
 drivers/clk/imx/clk-imx7d.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/clk/imx/clk-imx7d.c b/drivers/clk/imx/clk-imx7d.c
index 631e0ad6c4f0..f47f4f46c3c1 100644
--- a/drivers/clk/imx/clk-imx7d.c
+++ b/drivers/clk/imx/clk-imx7d.c
@@ -388,6 +388,7 @@ static int const clks_init_on[] __initconst = {
 	IMX7D_DRAM_PHYM_ROOT_CLK, IMX7D_DRAM_ROOT_CLK,
 	IMX7D_DRAM_PHYM_ALT_ROOT_CLK, IMX7D_DRAM_ALT_ROOT_CLK,
 	IMX7D_AHB_CHANNEL_ROOT_CLK, IMX7D_IPG_ROOT_CLK,
+	IMX7D_OCOTP_CLK
 };
 
 static struct clk_onecell_data clk_data;
-- 
2.17.1

